File: Week 1/Problem Set/mario.c
#include <cs50.h>
#include <stdbool.h>
#include <stdio.h>

int get_height(void);
void print_pyramid(int height_of_pyramid);
void print_a_collum(int row_length, int height_of_pyramid, bool is_left);
void print_the_spaces(int height_of_pyramid, int row_length);

int main(void)
{
    int height = get_height();

    print_pyramid(height);
}

int get_height(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    return height;
}

void print_pyramid(int height_of_pyramid)
{
    for (int row = 0; row < height_of_pyramid; row++)
    {

        print_a_collum(row, height_of_pyramid, true);

        printf("  ");

        print_a_collum(row, height_of_pyramid, false);

        printf("\n");
    }
}

void print_a_collum(int row_length, int height_of_pyramid, bool is_left)
{
    if (is_left)
    {
        print_the_spaces(height_of_pyramid, row_length);
    }

    for (int column = 0; column <= row_length; column++)
    {
        printf("#");
    }
}

void print_the_spaces(int height_of_pyramid, int row_length)
{
    for (int spaces = 0; spaces < height_of_pyramid - row_length - 1; spaces++)
    {
        printf(" ");
    }
}

File: Week 6/Problem Set/mario.py
def print_the_spaces(height_of_pyramid, row_length):
    for spaces in range(0, height_of_pyramid - row_length):
        if spaces != 0:
            print(" ", end="")


def print_a_column(row_length, height_of_pyramid, is_left):
    if is_left:
        print_the_spaces(height_of_pyramid, row_length)

    for column in range(row_length + 1):
        print("#", end="")


def print_pyramid(height_of_pyramid):
    for row in range(height_of_pyramid):
        print_a_column(row, height_of_pyramid, True)

        print("  ", end="")

        print_a_column(row, height_of_pyramid, False)

        print()


def get_height():
    while True:
        try:
            height = int(input("Height: "))
            if height >= 1 and height <= 8:
                return height
        except ValueError:
            pass


def main():
    height = get_height()

    print_pyramid(height)


if __name__ == "__main__":
    main()
